unity/Hello_World/Assets/Scripts/PVCameraCapture.cs

```diff
@@ -109,18 +122,68 @@ async void Start()
         }
 
         log.LogInfo("Media capture started");
+
+        try
+        {
+            worldOrigin = Marshal.GetObjectForIUnknown(UnityEngine.XR.WindowsMR.WindowsMREnvironment.OriginSpatialCoordinateSystem) as SpatialCoordinateSystem;
+            if (worldOrigin == null)
+            {
+                log.LogInfo("Unable to get world origin");
+            }
+        }
+        catch (Exception e)
+        {
+            log.LogInfo(e.ToString());
+        }
+
 #endif
     }
 
+    void Update()
+    {
+        if (debugString != "")
+        {
+            this.logger().LogInfo(debugString);
+            debugString = "";
+        }
+
+        projectionMatrix = Camera.main.projectionMatrix;
+    }
+
+    void SetupPVCapture()
+    {
+        try
+        {
+            IPAddress localAddr = IPAddress.Parse(TcpServerIPAddr);
+
+            // TcpListener server = new TcpListener(port);
+            tcpServer = new TcpListener(localAddr, PVTcpPort);
+
+            // Start listening for client requests.
+            tcpServer.Start();
+
+            // Perform a blocking call to accept requests.
+            // You could also use server.AcceptSocket() here.
+            tcpClient = tcpServer.AcceptTcpClient();
+            tcpStream = tcpClient.GetStream();
+        }
+        catch (Exception e)
+        {
+            debugString += e.ToString();
+        }
+    }
+
 #if ENABLE_WINMD_SUPPORT
+    // The following functions were adapted from:
+    // https://github.com/qian256/HoloLensARToolKit/blob/master/HoloLensARToolKit/Assets/ARToolKitUWP/Scripts/ARUWPVideo.cs
     public async Task<bool> InitializeMediaCaptureAsyncTask()
     {
         int targetVideoWidth, targetVideoHeight;
         float targetVideoFrameRate;
-        targetVideoWidth = 1280;
-        targetVideoHeight = 720;
-        //targetVideoWidth = 1920;
-        //targetVideoHeight = 1080;
+        //targetVideoWidth = 1280;
+        //targetVideoHeight = 720;
+        targetVideoWidth = 1920;
+        targetVideoHeight = 1080;

```

Comment:
Is this an arbitrary increase in resolution, or was this required for the "fix" to world localization? I'm not opposed, just curious / trying to understand things better.

Reply:
It was one of the things I tried to fix the localization issue. It didn't end up fixing the issue, but it didn't hurt so I left it in.